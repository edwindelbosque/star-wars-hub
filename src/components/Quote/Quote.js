import React from 'react';
import './Quote.scss';

const Quote = () => {
    return (
        <div className="galaxy">
            <div className="fall">
            <p>
                <i>EPISODE IV - A NEW HOPE</i>
                It is a period of civil war. Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire's ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet. 
                Pursued by the Empire's sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxyâ€¦.</p>    
            </div>
        </div>
    )
}

export default Quote