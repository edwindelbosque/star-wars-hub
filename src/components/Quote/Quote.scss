@keyframes swcrawl {
  0% {
    transform: rotatex(35deg) translateY(470px);
    opacity: 1;
  }
  25% {
    opacity:1;
  }
  90% {
    opacity:1;
  }
  100% {
    opacity: 0;
		transform: rotatex(45deg) translateY(-250px);
	}
}

body {                                        
  background-image: url('../../images/background-image.jpg');
  background-attachment: fixed;
  color: #E6CA3A;
}


.galaxy {
  display: flex;
  flex-direction: column;
  top: 0;
    
  div img {
    width: 80%;
    opacity: 0;
    animation-name: logo;
    animation-duration: 100s;
  }
    
  .fall {
    width: 600px;
    font-family: 'Franklin Gothic Book',sans-serif;
    font-size: 30px;
    line-height: 50px;
		margin: auto;
		margin-right: 80px;
    text-align: justify;
    perspective: 1150px;
    font-weight: bold;
        
    p {
      margin: auto;
      width: 100%;
      height: 100%;
      text-align: justify;
      animation-name: swcrawl;
      animation-delay: 0s;
      animation-duration: 20s;
      opacity: 1;
            
      i {
				text-align: center;
				color: rgb(255, 255, 255);
        display: block;
				width: 100%;
        font-size: 42px;
        margin-bottom: 30px;
      }
    }
  }
} 