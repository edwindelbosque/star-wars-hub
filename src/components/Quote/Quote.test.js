import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote', () => {
  it('component should match the snapshot', () => {
    const wrapper = shallow(
      <Quote
        episode={3}
        title={'Phantom Menace'}
        description={'Long description goes here..'}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });
})