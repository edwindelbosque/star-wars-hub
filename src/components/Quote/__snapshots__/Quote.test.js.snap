// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component should match the snapshot 1`] = `
<div
  className="galaxy"
>
  <div
    className="fall"
  >
    <p>
      <i>
        3
        . 
        PHANTOM MENACE
      </i>
    </p>
  </div>
</div>
`;
